### ISSUE TITLE
Loss callback in Training Flow

### ISSUE THREAD

**Issue by vlomonaco** [ContinualAI/avalanche#125]
It seems the current Training Flow is the following for the `MTDeepLearningStrategy`class:

```
  # - BeforeTraining (group)
  # --- set_initial_epoch
  # --- before_training
  # --- make_train_dataset
  # --- adapt_train_dataset
  # --- make_train_dataloader
  # - ModelTraining (group)
  # --- TrainingLoop (group)
  # ----- before_training_epoch
  # ----- TrainingEpoch (group)
  # ------- training_epoch
  # ----- after_training_epoch
  # --- TrainingModelAdaptation (group)
  # ----- set_task_layer
  # --- next_training_epoch
  # - AfterTraining (group)
  # --- after_training
```

So it seems I cannot define just the loss somewhere, but I need to write the entire `training_epoch ` function! We should add a method to customize the loss or add a regularization term as in the old training API.
